### PR TITLE
1428145: Do not deadlock on simultaneous hypervisor checkin

### PR DESCRIPTION
**Pull request by cnsnyder**
The tests for this PR will fail until our docker mysql images are updated to set the transaction isolation level to READ-COMMITTED.

The PR from the branch docker-refactor includes those changes.

### CODE REVIEW

**Comment by Ceiu** [comment]
@candlepin-pull-request-bot retest this please

File: server/src/main/java/org/candlepin/model/ConsumerCurator.java [hunk 1]
```diff
@@ -635,20 +635,19 @@ public VirtConsumerMap getHostConsumersMap(Owner owner,
...
-        List<Criterion> ors = new LinkedList<Criterion>();
+        LinkedList<String>  hypervisorIdsUpper = new LinkedList<String>();
         for (String hypervisorId : hypervisorIds) {
-            ors.add(Restrictions.eq("hvsr.hypervisorId", hypervisorId).ignoreCase());
-        }
-        return Restrictions.or(ors.toArray(new Criterion[0]));
+            hypervisorIdsUpper.add(hypervisorId.toUpperCase());
+        }
+        String query = "SELECT c FROM Consumer c WHERE " +
+                       "(c.owner.key = :ownerKey AND UPPER(c.hypervisorId.hypervisorId) IN (:hypervisorIds)) " +
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/model/ConsumerCurator.java]: We have to be careful with IN operations. Is it possible we'll hit the operator limit here?

**Comment by Ceiu** [comment]
@candlepin/jenkins-bot retest this please

File: server/src/main/java/org/candlepin/model/ConsumerCurator.java [hunk 2]
```diff
@@ -635,20 +627,33 @@ public VirtConsumerMap getHostConsumersMap(Owner owner,
...
-    private Criterion getHypervisorIdRestriction(Collection<String> hypervisorIds) {
-        List<Criterion> ors = new LinkedList<Criterion>();
+        LinkedList<String> hypervisorIdsUpper = new LinkedList<String>();
+        // Check for hypervisor ids case-insensitively
         for (String hypervisorId : hypervisorIds) {
-            ors.add(Restrictions.eq("hvsr.hypervisorId", hypervisorId).ignoreCase());
+            hypervisorIdsUpper.add(hypervisorId.toUpperCase());
+        }
+        List<Consumer> consumers = new LinkedList<Consumer>();
+        String query = "SELECT c FROM Consumer c WHERE " +
```
Comment (barnabycourt) [inline server/src/main/java/org/candlepin/model/ConsumerCurator.java]: Please add some descriptive comments here for history when someone is browsing the code later.

**Comment by cnsnyder** [comment]
@candlepin/jenkins-bot retest this please

**Comment by Ceiu** [comment]
ACK

**Pull request merged**